src/hotspot/share/runtime/arguments.cpp

```diff
@@ -553,6 +553,7 @@ static SpecialFlag const special_jvm_flags[] = {
   { "UseNewFieldLayout",             JDK_Version::jdk(15), JDK_Version::jdk(16), JDK_Version::jdk(17) },
   { "UseSemaphoreGCThreadsSynchronization", JDK_Version::undefined(), JDK_Version::jdk(16), JDK_Version::jdk(17) },
   { "ForceNUMA",                     JDK_Version::jdk(15), JDK_Version::jdk(16), JDK_Version::jdk(17) },
+  { "InsertMemBarAfterArraycopy",    JDK_Version::undefined(), JDK_Version::jdk(16), JDK_Version::jdk(17) },
```

Comment:
This seems like a product flag, so needs to follow `ForceNUMA` example? E.g. 16 -> 17 -> 18? And a provisional CSR?

Reply:
Since the flag is broken, I think immediate obsoletion, rather than deprecation makes more sense. See also discussion in JDK-8252889. A CSR has been created (JDK-8252890).

Reply:
Okay then!